### PR TITLE
Clean file type selection in new archive dialog

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

**Comment by rbuj** [comment]
Select Archive -> New... from menubar of Engrampa window. Click on `All Supported Files` to sow all supported files.

### Before

VirtualBox_Fedora 31 MATE - 9_02_10_2019_01_47_55

### After

VirtualBox_Fedora 31 MATE - 9_02_10_2019_01_54_21

**Comment by vkareh** [comment]
Thanks for the clarification

**Comment by rbuj** [comment]
The merge conflict has been resolved.

**Comment by rbuj** [comment]
The merge conflicts have been resolved.

File: src/eggfileformatchooser.c [hunk 1]
```diff
@@ -186,11 +186,11 @@ egg_file_format_filter_add_extensions (GtkFileFilter *filter,
...
       length = strcspn (extptr, ",");
-      pattern = g_new (gchar, length + 3);
+      pattern = g_new (gchar, length + 2);
 
-      memcpy (pattern, "*.", 2);
-      memcpy (pattern + 2, extptr, length);
-      pattern[length + 2] = '\0';
+      memcpy (pattern, "*", 1);
+      memcpy (pattern + 1, extptr, length);
+      pattern[length + 1] = '\0';
```
Comment (raveit65) [inline src/eggfileformatchooser.c]: We get a warning here.
```
eggfileformatchooser.c:193:27: warning: Out of bound memory access (access exceeds upper limit of memory block)

      pattern[length + 1] = '\0';

      ~~~~~~~~~~~~~~~~~~~~^~~~~~
```
https://travis-ci.org/mate-desktop/engrampa/jobs/592934779#L1289

**Comment by raveit65** [comment]
Beside from this warning the PR does what it promised and fixes the bug.

**Comment by rbuj** [comment]
@raveit65 I removed the build warning from eggfileformatchooser.c

**Comment by raveit65** [comment]
Thanks for fixing the build warning.

**Pull request merged**